### ISSUE TITLE
Dialog: Focus from input lost in very short time after dialog opened

### ISSUE THREAD

**Issue by martin654** [primefaces/primefaces#11860]
### Describe the bug

Hi, 
we have many dialogs in our app and our users were uset that focus was set inside the dialog when dialog opened - for example into first input component. That worked great in version 13.0.5 but since 13.0.6 this doesn't work anymore. I tried it with 13.0.9 and it is still broken.

In the browser you can see that focus is applied for short amount of time and then lost.

Please see attached reproducer.

Thanks

### Reproducer

Please see attached reproducer:  
[primefaces-test_dialog_focus.zip](https://github.com/primefaces/primefaces/files/15199180/primefaces-test_dialog_focus.zip)

### Expected behavior

I expect PF to work the same as in PF version 13.0.5 and earlier...

### PrimeFaces edition

Elite

### PrimeFaces version

13.0.6

### JSF implementation

Mojarra

### JSF version

4.0.7

### Java version

17

### Browser(s)

Chrome

**Comment by martin654** [comment]
Description in this ticket:
https://github.com/primefaces/primefaces/issues/11818 

"Since PF 13.0.6, setting the focus with the 'focus' attribute in the 'dialog' no longer works. The element briefly gains focus and then immediately loses it again. This still worked in PF 13.0.5." looked like it could be related

**Comment by martin654** [comment]
Looks like it is caused by this change:
https://github.com/primefaces/primefaces/commit/bc7bc9161597eb7563f2d55b80359097f622aa26

**Comment by melloware** [comment]
Thanks for the reproducer let me debug this issue.

**Comment by melloware** [comment]
@martin654 great reproducer this was tricky to debug.  What is happening is both dialogs called `show` before the first `hide` is executed and the delay is causing the focus change.  here is the debug.

```
applyFocus ajaxDialog
Start AJAX
applyFocus searchDialog
Complete AJAX
returnFocus 
<button id="frmTest:openDialogButton" 
```

Working on a fix

**Comment by martin654** [comment]
@melloware  Thank you for the fix, works great in 13.0.10.
